minitwit/src/main/java/WebApplication.java

```diff
@@ -665,16 +668,69 @@ else if (!BCrypt.checkpw(enteredPW, rs.getString("pw_hash"))) {
         updateLatest(request);
 
         var messages = getMessages();
-        var filteredMessages = messages.stream().map((message) -> {
-            return Map.ofEntries(
-                    Map.entry("content", message.get("text")),
-                    Map.entry("pub_date", message.get("pub_date")),
-                    Map.entry("user", message.get("username"))
-            );
-        });
+        var filteredMessages = messages.stream().map((message) -> Map.ofEntries(
+               Map.entry("content", message.get("text")),
+               Map.entry("pub_date", message.get("pub_date")),
+               Map.entry("user", message.get("username"))
+       ));
         return filteredMessages.toList();
     };
 
+    //  --------------------------------------
+    //              Simulator API
+    //  ---------------------------------------
+
+    public static Route serveSimMsgsUsr = (Request request, Response response) -> {
+
+        updateLatest(request);
+
+        SQLite db = new SQLite();
+        var connection = db.getConnection();
+        var userID = getUserID(db, request.params(":username"));
+
+        int noMsgs =  Integer.parseInt(request.queryParamOrDefault("no", "100"));
+
+        if (request.requestMethod().equals("GET")){
+            if (userID == 0) {
+                response.status(404);
+            }
+            else {
+                var query = connection.prepareStatement("SELECT message.*, user.* FROM message, user " +
+                        "                   WHERE message.flagged = 0 AND" +
+                        "                   user.user_id = message.author_id AND user.user_id = ?" +
+                        "                   ORDER BY message.pub_date DESC LIMIT ?" );
+                query.setInt(1, userID);
+                query.setInt(2, noMsgs);
+                var messages = query.executeQuery();
+                var filteredMessages = new ArrayList<Map>();
+                while(messages.next()) {
+                    var filteredMessage = new HashMap();
+                    filteredMessage.put("content", messages.getString("text"));
+                    filteredMessage.put("pub_date", messages.getInt("pub_date"));
+                    filteredMessage.put("user", messages.getString("username"));
+                    filteredMessages.add(filteredMessage);
+                }
+                connection.close();
+                return filteredMessages.stream().toList();
+            }
+        }
+        // Might pose a problem when we POST a message for a user that does not exist (lots of author_id=0 messages) not sure
+        else if (request.requestMethod().equals("POST")) {
+            var text = request.queryParams("content");
```

Comment:
The content should come from the body and not a query parameter, eg. `curl -X POST -H "Content-Type: application/json" -d '{"content": "Hello"}' --header "Authorization: Basic c2ltdWxhdG9yOnN1cGVyX3NhZmUh" http://localhost:8080/api/msgs/asdf?latest=123`